internal/cli/base.go

```diff
@@ -397,6 +399,95 @@ func (c *baseCommand) Init(opts ...Option) error {
 	return nil
 }
 
+// remoteIsPossible attempts to determine if the current waypoint infrastructure will be successful
+// performing a remote operation against this project. It verifies the project's datasource,
+// it's ODR runner profile, and detects if a remote runner is currently registered.
+func remoteIsPossible(ctx context.Context, client pb.WaypointClient, project *pb.Project, log hclog.Logger) (bool, error) {
+	// Check if remote is disabled in the waypoint.hcl
+
+	if !project.RemoteEnabled {
+		log.Debug("Remote operations are disabled for this project - operation cannot occur remotely")
+		return false, nil
+	}
+
+	if project.DataSource == nil {
+		log.Debug("Project has no datasource configured - operation cannot occur remotely")
+		// This is probably going to be fatal somewhere downstream
+		return false, nil
+	}
+
+	var hasRemoteDataSource bool
+	switch project.DataSource.GetSource().(type) {
```

Comment:
Should we be checking this in addition to the `RemoteEnabled` bool? https://github.com/hashicorp/waypoint/blob/7ba0386c24a98b3034c2fcb8dee9e397c12845e5/internal/server/proto/server.proto#L584

I think we have an explicit coupling on if a data source is git it's technically remote, but this this option is explicitly telling us if the project is remote or not...assuming it's correct and properly in all cases 🤔

Reply:
Yup, that's the first thing we check in this function!

Reply:
Oops, missed it! 😄